patches/api/0479-Fix-issues-with-recipe-API.patch

```diff
@@ -78,24 +87,88 @@ index 39f9766a03d420340d79841197f75c8b1dd49f4a..4e59f5176fd6cf92457ad750081c253a
          }
      }
 diff --git a/src/main/java/org/bukkit/inventory/RecipeChoice.java b/src/main/java/org/bukkit/inventory/RecipeChoice.java
-index db8bcc66bdc4bedfffb4705db6338eda4c0ad29a..c143b0d8b97d514566bac8413d0346cf50822aeb 100644
+index db8bcc66bdc4bedfffb4705db6338eda4c0ad29a..b8ad3ac02208343173e0001245f90507ae14cda7 100644
 --- a/src/main/java/org/bukkit/inventory/RecipeChoice.java
 +++ b/src/main/java/org/bukkit/inventory/RecipeChoice.java
-@@ -35,6 +35,13 @@ public interface RecipeChoice extends Predicate<ItemStack>, Cloneable {
+@@ -19,6 +19,46 @@ import org.jetbrains.annotations.NotNull;
+  */
+ public interface RecipeChoice extends Predicate<ItemStack>, Cloneable {
+ 
++    // Paper start - add "empty" choice
++    /**
++     * An "empty" recipe choice. Only valid as a recipe choice in
++     * specific places. Check the javadocs of a method before using it
++     * to be sure it's valid for that recipe and ingredient type.
++     *
++     * @return the empty recipe choice
++     */
++    static RecipeChoice empty() {
++        //<editor-fold desc="holder" defaultstate="collapsed">
++        final class Holder {
```

Comment:
I personally think having the usual package private class is much cleaner than this, at least in this case